### PR TITLE
Add `BASE_MODEL_NAME` key in `VISSLAdapter`

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/lightning-flash/issues/1068

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [ ] Did you write any **new necessary tests**? [not needed for typos/docs]
- [ ] Did you verify **new and existing tests pass** locally with your changes?
- [ ] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/image/embedding/vissl/adapter.py [hunk 1]
```diff
@@ -126,6 +126,7 @@ def from_task(
     def get_model_config_template():
         cfg = AttrDict(
             {
+                "BASE_MODEL_NAME": "multi_input_output_model",
```
Comment (krshrimali) [inline flash/image/embedding/vissl/adapter.py:129]: Is it okay if it defaults to `multi_input_output_model`?

**Comment by krshrimali** [comment]
Note to self: In a follow-up PR, we should also add tests to ensure bugs like https://github.com/PyTorchLightning/lightning-flash/issues/1068 are catched. What do you think? @ethanwharris

**Comment by ethanwharris** [comment]
@krshrimali Yeah, I think we could beef up the parametrisation of this test a bit to include dino: https://github.com/PyTorchLightning/lightning-flash/blob/a396e26f15b6cd2c12848102d878b63829bd842e/tests/image/embedding/test_model.py#L55

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Added
 
 - Added the normalization parameters of ```torchvision.transforms.Normalize``` as ```transform_kwargs``` in the ```ImageClassificationInputTransform``` ([#1178](https://github.com/PyTorchLightning/lightning-flash/pull/1178))
+- Added `BASE_MODEL_NAME` key in `VISSLAdapter`. ([#1184](https://github.com/PyTorchLightning/lightning-flash/pull/1184))
```
Comment (ethanwharris) [inline CHANGELOG.md:12]: Should this instead be a `Fixed` rather than added?

**Review by ethanwharris — COMMENTED** [review]
Looking good 😃 Let's also get a test, happy for you to add it in a follow-up if you prefer

File: flash/image/embedding/vissl/adapter.py (hunk 1 above)
Comment (krshrimali) [reply flash/image/embedding/vissl/adapter.py:129]: Okay, looks like this is fine. `vissl` default config also includes this: https://github.com/facebookresearch/vissl/blob/main/vissl/config/defaults.yaml#L504.

File: CHANGELOG.md (hunk 2 above)
Comment (krshrimali) [reply CHANGELOG.md:12]: I was also thinking about this, but it adds a key, and then it also fixes a bug. Uhmm, what do you prefer?

Comment (ethanwharris) [reply CHANGELOG.md:12]: I would go fixing :smiley: I try to use changelog as user facing changes (e.g. in this case "fixes support for dino in the image embedder" or something like that).

**Comment by krshrimali** [comment]
@ethanwharris - Thanks for reviewing the PR! I guess adding tests in this PR is a good idea, I'll try pushing the commits soon and request your review again. :))

File: CHANGELOG.md (hunk 2 above)
Comment (krshrimali) [reply CHANGELOG.md:12]: Makes sense, thanks @ethanwharris - I'll update this with the commit including tests once ready.

**Comment by krshrimali** [comment]
Superseded by https://github.com/PyTorchLightning/lightning-flash/pull/1196. Closing this :)

**Pull request closed without merging**